openfl-tutorials/experimental/Phi3/Workflow_Interface_Phi3.ipynb

```diff
@@ -0,0 +1,640 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "14821d97",
+   "metadata": {},
+   "source": [
+    "# Workflow Interface\n",
+    "## Fine-tuning Phi3 using OpenFL\n",
+    "[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/intel/openfl/blob/develop/openfl-tutorials/experimental/Phi3/Workflow_Interface_Phi3.ipynb)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "bd059520",
+   "metadata": {},
+   "source": [
+    "In this tutorial, we build on the ideas from the [first](https://github.com/intel/openfl/blob/develop/openfl-tutorials/experimental/Workflow_Interface_101_MNIST.ipynb) quick start notebook, and demonstrate how to fine-tune a Large Language Model (LLM) in a federated learning workflow. \n",
+    "\n",
+    "We will fine-tune **Microsoft's [Phi3](https://huggingface.co/docs/transformers/main/en/model_doc/phi3)** model on the [Math_10k](https://github.com/AGI-Edgerunners/LLM-Adapters/tree/main) dataset, an open-source mathematical question-answer pair dataset collected from multiple smaller math datasets."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "39c3d86a",
+   "metadata": {},
+   "source": [
+    "# What is it?"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a7989e72",
+   "metadata": {},
+   "source": [
+    "The workflow interface is a new way of composing federated learning expermients with OpenFL. It was borne through conversations with researchers and existing users who had novel use cases that didn't quite fit the standard horizontal federated learning paradigm. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "124ae236-2e33-4349-9979-f506d796276d",
+   "metadata": {},
+   "source": [
+    "### Installing OpenFL\n",
+    "- Lets now install OpenFL and the necessary dependencies for the workflow interface by running the cell below:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c808dd12-6795-4203-9221-0f6b43fc785f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install git+https://github.com/intel/openfl.git\n",
+    "!cd /home/oamontoy/workspace/openfl/openfl-tutorials/experimental/Phi3 && pip install -r ../requirements_workflow_interface.txt\n",
+    "!pip install numpy --upgrade\n",
+    "!pip install transformers peft datasets trl"
```

Comment:
when running "!pip install transformers peft datasets trl" getting the below error
```
      -- Running cmake for PyArrow
      cmake -DCMAKE_INSTALL_PREFIX=/private/var/folders/bj/xlf52n2s7zx24gvllky2f_wh0000gn/T/pip-install-55o676u7/pyarrow_50ccfaecb1b946ff8ad649bdf24d7546/build/lib.macosx-10.9-x86_64-cpython-311/pyarrow -DPYTHON_EXECUTABLE=/Users/rkrishne/anaconda3/bin/python -DPython3_EXECUTABLE=/Users/rkrishne/anaconda3/bin/python -DPYARROW_CXXFLAGS= -DPYARROW_BUNDLE_ARROW_CPP=off -DPYARROW_BUNDLE_CYTHON_CPP=off -DPYARROW_GENERATE_COVERAGE=off -DCMAKE_BUILD_TYPE=release /private/var/folders/bj/xlf52n2s7zx24gvllky2f_wh0000gn/T/pip-install-55o676u7/pyarrow_50ccfaecb1b946ff8ad649bdf24d7546
      error: command 'cmake' failed: No such file or directory
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for pyarrow
Failed to build pyarrow
ERROR: Could not build wheels for pyarrow, which is required to install pyproject.toml-based projects
```

Reply:
to resolve tat error we need to install cmake package but even after installing cmake package getting the below error
`/private/var/folders/bj/xlf52n2s7zx24gvllky2f_wh0000gn/T/pip-install-doldjp2l/pyarrow_53bf7158da274fff8d32761768c8e19b/pyarrow/src/arrow/python/arrow_to_pandas.cc:208:16: error: no member named 'LARGE_LIST_VIEW' in 'arrow::Type'; did you mean 'LARGE_LIST'?
          case Type::LARGE_LIST_VIEW: {
               ~~~~~~^~~~~~~~~~~~~~~
                     LARGE_LIST
      /anaconda3/include/arrow/type_fwd.h:410:5: note: 'LARGE_LIST' declared here
          LARGE_LIST = 36,
          ^
      fatal error: too many errors emitted, stopping now [-ferror-limit=]
      20 errors generated.
      make[2]: *** [CMakeFiles/arrow_python.dir/pyarrow/src/arrow/python/arrow_to_pandas.cc.o] Error 1
      make[1]: *** [CMakeFiles/arrow_python.dir/all] Error 2
      make: *** [all] Error 2
      error: command '/usr/local/bin/cmake' failed with exit code 2
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for pyarrow
Failed to build pyarrow
ERROR: Could not build wheels for pyarrow, which is required to install pyproject.toml-based projects
Note: you may need to restart the kernel to use updated packages.`